docs/docs/how-to/implement_recognizer.md

```diff
@@ -1 +1,305 @@
+is why they can be used as root checks.
+
+### The first Check
+
+The first thing we want to look for when trying to detect Singleton implementations, is a class. So
+lets add a check for that:
+
+```csharp
+IEnumerable< ICheck > IRecognizer.Create()
+{
+    yield return Class(
+        Priority.Knockout
+    );
+}
+```
+
+Lets go over the things we see in this implementation. The first thing you may notice, is that
+`Create` returns an `IEnumerable< ICheck >`, and that the root check is returned using a `yield
+return`. The reason you can have multiple root checks, is that there are design patterns which
+consist of multiple classes and interfaces. By returning you multiple root checks, you can match all
+of those using the same recognizer. In a later section we will also show how to check for relations
+between these entities. The second important thing is the priority, which is `Knockout` in this
+case. Using priorities you can differentiate between different requirements, and assign an
+importance to them. The `Knockout` priority is special in the sense that this marks the check as
+required, if PatternPal sees that a check has failed, but it is marked as `Knockout`, it will
+immediately disregard the entity being checked from being a possible match.
+
+## Adding nested Checks
+
+Now that you have seen how to create a check, it is time to learn how to add a nested check. A
+nested check is a check which is the child of another check. An example of this is a `MethodCheck`
+which is the child of a `ClassCheck`.
+
+### Private Constructor Check
+
+To be a Singleton implementation, the class must have a private constructor, so lets add a check
+for that.
+
+```csharp
+IEnumerable< ICheck > IRecognizer.Create()
+{
+    yield return Class(
+        Priority.Knockout,
+        Constructor(
+            Priority.Knockout,
+            Modifiers(
+                Priority.Knockout,
+                Modifier.Private
+            )
+        )
+    );
+}
+```
+
+In this example we immediately see two levels of nesting: the first nested check is the
+`ConstructorCheck`, and the second nested check is the `ModifierCheck` inside the
+`ConstructorCheck`. This recognizer will now match any class which has a private constructor.
+
+## Operator Checks
+
+As you may have noticed in the previous section, the recognizer does not disallow any non-private
+constructors, which is an issue in case of a singleton implementation. To check for the absence of
+something, you can use the `NotCheck`.
+
+```csharp
+IEnumerable< ICheck > IRecognizer.Create()
+{
+    yield return Class(
+        ...
+        Constructor(
+            Priority.Knockout,
+            Not(
+                Priority.Knockout,
+                Modifiers(
+                    Priority.Knockout,
+                    Modifier.Public
+                )
+            )
+        )
+    );
+}
+```
+
+Because a constructor can have only one modifier, you cannot put the `NotCheck` inside the
```

Comment:
Unclear this refers to the number of access modifiers something in c# can have, instead of the amount that can be present in a Check. Multiple modifiers could be present in a ConstructorCheck by for example putting them in an Any or All check.